### DISCUSSION TITLE
Optional enforcement of prefix

### DISCUSSION THREAD

**Discussion by mreed5168** [ben-rogerson/twin.macro#622] [Ideas]
Curious to get thoughts on making the enforcement of the TW prefix optional. I'd be willing to open a PR for this, with updated tests, docs, etc...

I use TW in Laravel blades and the prefix is necessary to prevent collisions at the blade level. I also use `twin.macro` with a React app that renders within the blades, and collisions aren't a problem. Although the latest versions of `twin.macro` throws an error when a prefix is defined in the TW config but not used with twin, the actual class is still resolved. I'd prefer not needing to update an entire component library with prefixes as enforcement could be a twin config option.

**Comment by mreed5168** [comment]
Any thoughts on this @ben-rogerson?

**Reply by ben-rogerson** [reply]
Adjusting the config to remove the prefix isn't an option in your case?

```diff
// tailwind.config.js
module.exports = {
-  prefix: 'tw-',
+  prefix: '',
}
```

**Reply by mreed5168** [reply]
Unfortunately not. It's a large monorepo with multiple sites and a component library, so I still need the prefix to prevent collisions on some  sites that use Tailwind and other libraries with matching class names.

**Reply by ben-rogerson** [reply]
I'm not too keen on adding a feature which disables another feature that is controllable in the tailwind config.  

Can you use the [presets feature](https://tailwindcss.com/docs/presets#extending-multiple-presets) to extend your base tailwind config and customise the prefix on your chosen sites?
